### PR TITLE
[SYCL][Doc] Document sycl.pi.debug XPTI stream

### PR DESCRIPTION
**Pull request by alexbatashev**
Also fix minor rendering issues

### CODE REVIEW

**Comment by alexbatashev** [comment]
@tovinkere together with https://github.com/intel/llvm/pull/4340 this will fix rendering of this page https://intel.github.io/llvm-docs/SYCLInstrumentationUsingXPTI.html

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 1]
```diff
@@ -227,16 +227,23 @@ by the SYCL runtime.
...
-| ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| Trace Point Type | Parameter Description | Metadata |
+| :--------------: | :-------------------- | :------- |
```
Comment (tovinkere) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md:231]: Looks like clang-format will add additional "---" to match field width, something to look out for. Thanks!

**Review by tovinkere — APPROVED** [review]
LGTM

File: sycl/doc/SYCLInstrumentationUsingXPTI.md (hunk 1 above)
Comment (bader) [reply sycl/doc/SYCLInstrumentationUsingXPTI.md:231]: I think `clang-format` is applicable for C-based programming languages, so Markdown documents should be ignored.

**Review by bader — DISMISSED** [review]

File: sycl/doc/SYCLInstrumentationUsingXPTI.md [hunk 2]
```diff
@@ -227,16 +227,23 @@ by the SYCL runtime.
...
 
+## SYCL Stream `"sycl.pi.debug"` Notification Signatures
+
+| Trace Point Type | Parameter Description | Metadata |
+| :--------------: | :-------------------- | :------- |
+| `function_begin` | <li>**trace_type**: `xpti::trace_point_type_t::function_begin` that marks the beginning of a function</li> <li> **parent**: Event ID created for all functions in the `sycl.pi.debug` layer.</li> <li> **event**: `nullptr` - since the stream of data just captures functions being called.</li> <li> **instance**: Unique ID to allow the correlation of the `function_begin` event with the `function_end` event. </li> <li> **user_data**: A pointer to `function_with_args_t` object, that includes function ID, name, and arguments. </li>                                                                                                                 | None     |
```
Comment (tovinkere) [inline sycl/doc/SYCLInstrumentationUsingXPTI.md]: One small miss - the trace point type should be "function_with_args_begin"/"function_with_args_end" for this documentation.

**Review by tovinkere — CHANGES_REQUESTED** [review]

**Review by tovinkere — APPROVED** [review]
LGTM

**Review by bader — APPROVED** [review]

**Pull request merged**